crypto/bio/bss_dgram_pair.c

```diff
@@ -452,15 +538,15 @@ static size_t dgram_pair_ctrl_get_write_guarantee(BIO *bio)
 /* BIO_dgram_get_local_addr_cap (BIO_CTRL_DGRAM_GET_LOCAL_ADDR_CAP) */
 static int dgram_pair_ctrl_get_local_addr_cap(BIO *bio)
 {
-    struct bio_dgram_pair_st *b = bio->ptr, *peerb;
-
-    if (b->peer == NULL)
-        return 0;
+    struct bio_dgram_pair_st *b = bio->ptr, *readb;
 
-    peerb = b->peer->ptr;
+    if (b->peer != NULL)
```

Comment:
Instead of adding another structure member I could check `bio->init`. I.e. - > `bio->init == 0 -> error` otherwise `b->peer` distinguishes between dgram_pair and dgram_mem. Would you be OK with that?

Reply:
Sounds good, but I'd suggest factoring that expression into a simple inline function, like `is_dgram_mem` or similar.